src/AI-DataFrameInspector/AISpDataFramePresenter.class.st

```diff
@@ -0,0 +1,250 @@
+				   yourself)
+		  height: 20;
+		  yourself;
+		  add: dataFramePresenter;
+		  yourself
+]
+
+{ #category : #editing }
+AISpDataFramePresenter >> editButtonAction [
+
+	dataFrameColumnsEditable := dataFrameColumnsEditable not.
+	dataFrameColumnsEditable
+		ifTrue: [
+			editButton
+				label: 'read';
+				icon: (self iconNamed: #book) ]
+		ifFalse: [
+			editButton
+				label: 'edit';
+				icon: (self iconNamed: #editIcon) ].
+	self editDataFrame.
+	self layout replace: dataFramePresenter with: newdataFramePresenter.
+	dataFramePresenter := newdataFramePresenter
+]
+
+{ #category : #editing }
+AISpDataFramePresenter >> editDataFrame [
+
+	newdataFramePresenter := self newTable.
+
+	newdataFramePresenter addColumn: (SpIndexTableColumn new
+			 title: '#';
+			 sortFunction: #yourself ascending;
+			 beNotExpandable;
+			 yourself).
+	dataFrame rowNames
+	= (1 to: dataFrame numberOfRows) asOrderedCollection ifFalse: [
+		newdataFramePresenter addColumn: (SpStringTableColumn
+				 title: ''
+				 evaluated: [ :rowWithName | rowWithName at: 1 ]) ].
+	dataFrameColumnsEditable
+		ifFalse: [
+			dataFrame columnNames doWithIndex: [ :columnName :idx |
+				newdataFramePresenter addColumn: (SpStringTableColumn new
+						 title: columnName;
+						 beSortable;
+						 evaluated: [ :selection |
+							 selection dataSeriesElementAt: idx + 1 ]) ] ]
+		ifTrue: [
+			dataFrame columnNames doWithIndex: [ :columnName :idx |
+				| column |
+				column := SpStringTableColumn
+					          title: columnName
+					          evaluated: [ :rowWithName |
+						          | originalValue |
+						          originalValue := rowWithName at: idx + 1.
+						          originalValue asString ].
+				column beEditable.
+				column onAcceptEdition: [ :row :newValue |
+					| originalRowName rowInd putVal |
+					putVal := newValue.
+					originalRowName := row at: 1.
+					rowInd := dataFrame indexOfRowNamed: originalRowName.
+					(dataFrame at: rowInd at: idx) isNumber ifTrue: [
+						putVal := newValue asNumber ].
+					((dataFrame at: rowInd at: idx) isKindOf: Boolean) ifTrue: [
+						newValue = 'true' ifTrue: [ putVal := true ].
+						newValue = 'false' ifTrue: [ putVal := false ] ].
+
+					dataFrame at: rowInd at: idx put: putVal ].
+				newdataFramePresenter addColumn: column ] ].
+	newdataFramePresenter beResizable.
+
+	newdataFramePresenter items: dataFrame asArrayOfRowsWithName
+]
+
+{ #category : #searching }
+AISpDataFramePresenter >> filterTableWithSearchQuery: newValue [
+
+	| filteredRows |
+	filteredRows := dataFrame select: [ :row |
+		                row anySatisfy: [ :value |
+			                value asString
+				                includesSubstring: newValue asString
+				                at: 1 ] ].
+		
+	newdataFramePresenter items: filteredRows asArrayOfRowsWithName
+]
+
+{ #category : #initialization }
+AISpDataFramePresenter >> initialize [
+
+	dataFrameColumnsEditable ifNil: [ dataFrameColumnsEditable := false ].
```

Comment:
Is the nil check useful? Can it be set before the #initialize method? If not, we should remove the guard

Reply:
No, this nil check is not useful as it won't be set before #initialize. I removed it now